packages/components/multiselect/src/Multiselect.tsx

```diff
@@ -172,43 +215,33 @@ function _Multiselect(props: MultiselectProps, ref: React.Ref<HTMLDivElement>) {
   }, [currentSelection, placeholder, styles.currentSelection]);
 
   const childrenLength = useMemo(
-    () => React.Children.count(children),
+    () =>
+      countMatchingChildren(
+        children,
+        (child) => child.type === MultiselectOption,
+      ),
     [children],
   );
 
   // clones and enriches the multiselect options
   const enrichOptions = React.useCallback(
     (children: React.ReactNode): React.ReactNode => {
-      return React.Children.map(children, (child) => {
-        if (React.isValidElement(child)) {
-          if (
-            child.type === React.Fragment ||
-            child.type === 'div' ||
-            child.type === 'ul'
-          ) {
-            return enrichOptions(child.props.children);
-          }
-          if (child.type === MultiselectOption) {
-            const onSelectItem = (
-              even: React.ChangeEvent<HTMLInputElement>,
-            ) => {
-              // Selecting an item triggers a rerendering and thereby losing the
-              // focus on the clicked element. To avoid having the focus on the document body
-              // (which breaks `closeOnBlur`), we force it back to the list in the popup.
-              internalListRef.current?.focus();
-              child.props?.onSelectItem(even);
-            };
-            return React.cloneElement(child, {
-              searchValue,
-              onSelectItem,
-            } as Partial<MultiselectOptionProps>);
-          }
-          return child;
-        }
-        return child;
-      });
+      return iterateOverChildren(
+        children,
+        (child) => child.type === MultiselectOption,
+        (child) => {
+          const onSelectItem = (even: React.ChangeEvent<HTMLInputElement>) => {
+            focusList();
+            child.props?.onSelectItem(even);
```

Comment:
```suggestion
          const onSelectItem = (evt: React.ChangeEvent<HTMLInputElement>) => {
            focusList();
            child.props?.onSelectItem(evt);
```

idk if it was my typo but even is really confusing me as variable name in this context

Reply:
True, I don't where this comes from ^^
I'd rather call it `event` since it doesn't save space but is more readable